Zend/zend_vm_def.h

```diff
@@ -7210,6 +7210,10 @@ ZEND_VM_HANDLER(149, ZEND_HANDLE_EXCEPTION, ANY, ANY)
 
 	i_cleanup_unfinished_execution(execute_data, op_num, catch_op_num);
 
+	if (EG(exception) && !instanceof_function(EG(exception)->ce, zend_ce_throwable)) {
+		finally_op_num = 0;
```

Comment:
Why? I thought it was part of the purpose to support proper shutdown including dtors and finally?